x86 build broken

rtpaint/layers/Layer.cpp: In static member function 'static Layer* Layer::readLayer(BFile&, ImageView*, int32, bool, int32)':
artpaint/layers/Layer.cpp:496:41: error: cannot convert 'uint64*' {aka 'long long unsigned int*'} to 'uLongf*' {aka 'long unsigned int*'}
  496 |                                         &actual_length,
      |                                         ^~~~~~~~~~~~~~
      |                                         |
      |                                         uint64* {aka long long unsigned int*}
In file included from /boot/system/develop/headers/x86/zlib.h:34,
                 from artpaint/layers/Layer.cpp:31:
/boot/system/develop/headers/x86/zlib.h:1267:32: note:   initializing argument 2 of 'int uncompress(Bytef*, uLongf*, const Bytef*, uLong)'
 1267 | ZEXTERN int ZEXPORT uncompress OF((Bytef *dest,   uLongf *destLen,
      |                                ^~
artpaint/layers/Layer.cpp: In member function 'int64 Layer::writeLayer(BFile&, int32)':
artpaint/layers/Layer.cpp:639:25: error: cannot convert 'uint64*' {aka 'long long unsigned int*'} to 'uLongf*' {aka 'long unsigned int*'}
  639 |                         &dataLengthCompressed,
      |                         ^~~~~~~~~~~~~~~~~~~~~
      |                         |
      |                         uint64* {aka long long unsigned int*}
In file included from /boot/system/develop/headers/x86/zlib.h:34,
                 from artpaint/layers/Layer.cpp:31:
/boot/system/develop/headers/x86/zlib.h:1229:30: note:   initializing argument 2 of 'int compress(Bytef*, uLongf*, const Bytef*, uLong)'
 1229 | ZEXTERN int ZEXPORT compress OF((Bytef *dest,   uLongf *destLen,
      |                              ^~

--- a/artpaint/layers/Layer.cpp
+++ b/artpaint/layers/Layer.cpp
@@ -486,14 +486,17 @@ Layer::readLayer(BFile& file, ImageView* imageView, int32 new_id,
 
 			if (compression_method == ZLIB_COMPRESSION) {
 				uint64 actual_length;
-				if (file.Read(&actual_length, sizeof(int64)) != sizeof(int64))
+				if (file.Read(&actual_length, sizeof(int64))
+					!= sizeof(int64))
 					return NULL;
 
 				uint8* uncompressedBits = (uint8*)malloc(actual_length);
+				unsigned long uncompressedLength =
+					static_cast<unsigned long>(actual_length);
 
 				int z_result = uncompress(
 					uncompressedBits,
-					&actual_length,
+					&uncompressedLength,
 					compressedBits,
 					old_length
 				);
@@ -631,17 +634,20 @@ Layer::writeLayer(BFile& file, int32 compression_method)
 	int z_result = Z_OK;
 
 	if (compression_method == ZLIB_COMPRESSION) {
-		uint64 dataLengthCompressed = (data_length * 1.1) + 12;
+		unsigned long dataLengthCompressedULong =
+			(data_length * 1.1) + 12;
 
-		uint8* dataCompressed = (uint8*)malloc(dataLengthCompressed);
+		uint8* dataCompressed = (uint8*)malloc(dataLengthCompressedULong);
 		z_result = compress(
 			dataCompressed,
-			&dataLengthCompressed,
+			&dataLengthCompressedULong,
 			(uint8*)fLayerData->Bits(),
-			data_length
+			static_cast<unsigned long>(data_length)
 		);
 
 		if (z_result == Z_OK) {
+			int64 dataLengthCompressed = static_cast<int64>(dataLengthCompressedULong);
+
 			written_bytes += file.Write(&dataLengthCompressed, sizeof(int64));
 			written_bytes += file.Write(dataCompressed, dataLengthCompressed);
 		} else {

